buildchain/buildchain/targets/remote_image.py

```diff
@@ -90,17 +90,34 @@ def _show(task: types.Task) -> str:
         return utils.title_with_target1('IMG PULL', task)
 
     def _build_actions(self) -> List[types.Action]:
+        """Compute actions for the image - pull, tag, save."""
         filepath = self.uncompressed_filename
+        pull_command = PullCommand(
+            repository=self.repository, digest=self.digest
+        )
+        if self.for_containerd:
+            repository = '{img.registry}/{img.name}'.format(img=self)
+        else:
+            repository = self.name
+        tag_command = TagCommand(
+            repository=repository, fullname=self.fullname, version=self.version
+        )
+
         actions : List[types.Action] = [
-            [config.DOCKER, 'pull', self.fullname],
-            [config.DOCKER, 'tag', self.fullname, self.tag],
-            [config.DOCKER, 'save', self.tag, '-o', str(filepath)],
+            (pull_command, [], {}),
+            (tag_command, [], {}),
+            (self.save_image, [], {'save_path': filepath})
         ]
         # containerd doesn't support compressed images.
         if not self.for_containerd:
             actions.append((coreutils.gzip, [filepath], {}))
         return actions
 
+    @property
+    def repository(self) -> str:
```

Comment:
Need to clarify the difference between the `repository` property and the `repository` computed above: it's really confusing.

Also, shouldn't the term be `registry` instead of `repository`?

Reply:
The terms are confusingly used in the API docs too. 

As far as I understand it, the repository is the layer below the registry. Dockerhub would be a registry, and `dockerhub/calico` would be the `calico` repository, containing every calico image.

I need to, and will, make a pass on how we refer to different items in these classes. Tags, versions, repositories, registries... are somewhat conflated in the code right now.

Reply:
Would be a good thing to call a cat a cat, but this renaming should be done in a dedicated PR (this one already big enough, and renaming is not it's goal).

For this PR, just make sure that there is no ambiguity between this property and what you compute above: that will be enough.

Reply:
Tracked in #947